exporter.go

```diff
@@ -40,26 +53,26 @@ func (e *Exporter) Export(
 	var err error
 
 	meta := metadata.LayersMetadata{}
-
 	meta.RunImage.TopLayer, err = workingImage.TopLayer()
 	if err != nil {
 		return errors.Wrap(err, "get run image top layer SHA")
 	}
 
+	// TODO why are we saving stack in addition to runImageRef when the two can be different?
```

Comment:
Question to the reviewers. Since metadata.Stack only contains the run image which may or may not have been used, why is it saved in addition to the actual run image in metadata.RunImage?

Reply:
It looks like `metadata.Stack.RunImage` is the set of (intentionally mutable) run image tags (including mirrors) that the rebaser should consider when selecting a run image based on the target registry during the rebase. 

`metadata.RunImageMetadata` is information about the specific run image that was used during the export.

Reply:
Does that mean that if I export my application with a custom run image through the -image flag and then run rebaser on the image, it will default to rebasing the app image on top of the stack run image? That seems like it could lead to unexpected behavior.

If we save only one of metadata.RunImage or metadata.Stack, rebaser can raise an error when the '-image'  flag isn't provided and the app is based on a custom run image.

Reply:
@lukasberger I'm not sure I understand what you are proposing. How would the `rebaser` know if the image was based on a custom run image w/o knowing both the specific run-image that was used and the default mutable run-image and mirror tag references?

If a user of the lifecycle want to change the default stack for rebase it can provide a `stack.toml` file to the `exporter` to set the `metadata.Stack.RunImage`. If this stack metadata is left empty, `rebaser` requires an `-image` flag. Right now both the specific base image to use and the metadata to set for rebase are independently configurable, which I believe gives platforms the most flexibility.

Reply:
The `rebaser` would know that the app image was based on a custom run image (specified through `-image` on `exporter`) if there was no stack specified in the metadata. It would raise an error when attempting to rebase such app image without also providing the new image to rebase to through `-image`.

What is the intended use case for exporting with image X as the base and setting image Y as the rebase target? Why not use Y from the start if the app image can reasonably be expected to end up with Y as the base at some point anyway?

Reply:
Resolving conversation, discussed offline and working as intended.